usl_models/usl_models/flood_ml/model.py

```diff
@@ -68,57 +67,11 @@ def _build_model(self) -> keras.Model:
         )
         return model
 
-    def _validate_and_preprocess_data(
-        self,
-        data: FloodModelData,
-        training=True,
-    ) -> FloodModelData:
-        """Validates model data and does all necessary preprocessing.
-
-        Args:
-            data: A FloodModelData object. Labels are required for training.
-            training: Whether data is used for model training. If True, labels
-                will be validated.
-
-        Returns:
-            A processed FloodModelData object.
-        """
-        if training:
-            # Labels are required for training.
-            if data.labels is None:
-                raise ValueError("Labels must be provided during model training.")
-
-            # Labels must match the storm duration.
-            expected_label_shape = list(data.labels.shape)
-            expected_label_shape[1] = data.storm_duration
-            assert data.labels.shape[1] == data.storm_duration, (
-                "Provided labels are inconsistent with storm duration. "
-                f"Labels are expected to have shape {expected_label_shape}. "
-                f"Actual shape: {data.labels.shape}."
-            )
-
-        # Check whether the temporal data is already windowed. If it is, checks
-        # the expected shape. Otherwise, create the window view.
-        if tf.rank(data.temporal) == 3:  # windowed: (B, T_max, m)
-            assert data.temporal.shape[-1] == self._model_params["m_rainfall"], (
-                "Mismatch between the temporal data window size "
-                f"({data.temporal.shape[-1]}) and the expected window size "
-                f"(m = {self._model_params['m_rainfall']})."
-            )
-        else:
-            full_temp_input = data_utils.temporal_window_view(
-                data.temporal, self._model_params["m_rainfall"]
-            )
-            data = dataclasses.replace(data, temporal=full_temp_input)
-
-        # We assume that, if provided, this input is a *single* flood map.
-        st_input = data.spatiotemporal
-        if st_input is None:
-            st_shape = data.geospatial.shape[:3] + [1]
-            st_input = tf.zeros(st_shape)
-
-        data = dataclasses.replace(data, spatiotemporal=st_input)
-        return data
+    def _load_weights_from_artifact(self, artifact_uri: str):
+        """Loads model weights from a GCS artifact URI."""
+        prediction_utils.download_model_artifacts(artifact_uri)
+        loaded_model = keras.models.load_model("model")
+        self._model.set_weights(loaded_model.get_weights())
```

Comment:
Is it just the weights we need? Or do we need `self._model = loaded_model` ?

Reply:
Loading the weights is the only way to keep the custom keras.Model class intact due to the aforementioned changes upon export.

Reply:
okay, some comments describing these issues would be helpful

this means that if the model architecture changes, old models essentially become un-loadable, right? Because they'd no longer be compatible with what `self._build_model()` produces?

Reply:
Sure, will document this.

Reply:
Am I correct that this makes old models un-loaded-able if the architecture changes? That seems like a problem for experimentation.
Is it just the call_n signature thing we need to fix to make this work?

Reply:
Under this setup, models will need to be loaded from the same version of the code they were saved in.
We could try moving call_n logic into the wrapper class. Then the only remaining issue is unwrapping the inputs dictionary into kwargs to match what is expected from serving_default.

Reply:
Okay, let's just continue focusing on documenting issues rather than solving issues. As long as we make clear what issues are and suggest some path forward for addressing them, I think we're leaving things in a good place.